File: Algo/Easy/0704_Binary_Search/Solution.py
from typing import *

## 2021-11-08
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0;
        right = len(nums) - 1
        
        if nums[left] == target:
            return left
        if nums[right] == target:
            return right
        
        while left < right - 1:
            mid = int((left + right) / 2)
            
            if nums[mid] > target:
                right = mid
            elif nums[mid] < target:
                left = mid
            else:
                return mid
        
        return -1

## 2022-08-29
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left, right = 0, len(nums)-1
        while left <= right:
            mid = left + (right-left)//2
            
            if nums[mid] > target:
                right = mid-1
            elif nums[mid] < target:
                left = mid+1
            else:
                return mid
        
        return -1

File: Algo/Easy/0704_Binary_Search/Solution.java
// 2021-08-01
class Solution {
    public int search(int[] nums, int target) {
        int low = 0;
        int high = nums.length - 1;
        if (nums[low] == target) return low;
        if (nums[high] == target) return high;
        
        while (high - low > 1) {
            int mid = (low + high) / 2;
            if (nums[mid] == target) {
                return mid;
            } else {
                if (nums[mid] < target) {
                    low = mid;
                } else {
                    high = mid;
                }
            }
        }
        
        return -1;
    }
}